mypy/checkpattern.py

```diff
@@ -244,6 +244,27 @@ def visit_sequence_pattern(self, o: SequencePattern) -> PatternType:
         # get inner types of original type
         #
         unpack_index = None
+
+        if isinstance(current_type, UnionType):
+            union_captures: dict[Expression, Type] = {}
+            union_items: list[Type] = []
+            for t in current_type.items:
+                typ, _, capture = self.accept(o, t)
+                if not is_uninhabited(get_proper_type(typ)):
+                    union_items.append(typ)
+                    union_captures.update(capture)
```

Comment:
hmm shouldn't the values in `union_captures` be a union of all the things, instead of potentially clobbering?